R/metadata-extraction_imagery_dataset.R

```diff
@@ -45,31 +51,31 @@ extract_mission_polygon = function(exif, image_merge_distance, min_contig_area =
   # Check if multipolygon and if so, keep only the poly parts > min aea, and if any removed, return
   # warning of how many removed
   if (n_polys > 1) {
-
     parts = sf::st_cast(poly, "POLYGON")
     areas = sf::st_area(parts)
     max_area = max(areas)
     parts_keep_idx = which(areas == max_area | areas > units::set_units(min_contig_area, "m2"))
     parts_filtered = parts[parts_keep_idx]
-    poly = parts_filtered |> sf::st_union() |> sf::st_cast("MULTIPOLYGON")
+    poly = parts_filtered |>
+      sf::st_union() |>
+      sf::st_cast("MULTIPOLYGON")
 
     n_polys_filtered = length(parts_filtered)
+    # TODO it's possible all polygons could be filtered out in this step
 
     if (n_polys_filtered < n_polys) {
       warning(n_polys, " non-contiguous image clusters in dataset ", exif$dataset_id[1], ". Retaining only the ", n_polys_filtered, " clusters with area > ", min_contig_area, " m^2.")
     } else {
       warning(n_polys, " non-contiguous image clusters in dataset ", exif$dataset_id[1], ". Retaining all becaus all have area > ", min_contig_area, " m^2.")
     }
-
   } else if (n_polys == 0) {
     stop("No contiguous images in dataset ", exif$dataset_id[1])
   }
 
 
-  polysimp = sf::st_simplify(poly, dTolerance = 10) |> sf::st_cast("MULTIPOLYGON")
+  polysimp = sf::st_simplify(poly, dTolerance = 1) |> sf::st_cast("MULTIPOLYGON")
```

Comment:
What is the reasoning behind this change? Is this the best value to use also if the method is specified as `dilate_erode`?

Reply:
Sorry, this was a one-off and I shouldn't have committed it. I got an issue with some small flight polygons being entirely eliminated by the value of 10 so I decreased it. Maybe there's a better way to preserve small polygons. 

Is this simplification primarily for computational reasons or to make the result look good? If it's computational, I wonder if a smaller simplification threshold such as 1 would still result in fast enough operations using the resultant polygon.

Reply:
It was more because I thought it was the best at capturing the polygon we actually designed the mission to represent, rather than (at finer scale) the idiosyncrasies of how it was actually collected.

However that may not be good logic (maybe we do want to capture the actual, not intended, polygon), and I didn't know that 10 could entirely eliminate the polygon, so I'd support using a smaller value.